webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/servers/PolicyServerManagementService.scala

```diff
@@ -140,6 +153,36 @@ class PolicyServerManagementServiceImpl(
       networks
     }
   }
+
+  /**
+   * Delete things related to a relay:
+   * - group: nodeGroupId=hasPolicyServer-${uuid},groupCategoryId=SystemGroups,groupCategoryId=GroupRoot,ou=Rudder,cn=rudder-configuration
+   * - rule target:  ruleTarget=policyServer:${RELAY_UUID},groupCategoryId=SystemGroups,groupCategoryId=GroupRoot,ou=Rudder,cn=rudder-configuration
+   * - directive:  directiveId=${RELAY_UUID}-distributePolicy,activeTechniqueId=distributePolicy,techniqueCategoryId=Rudder Internal,techniqueCategoryId=Active Techniques,ou=Rudder,cn=rudder-configuration
+   * - directive: directiveId=common-${RELAY_UUID},activeTechniqueId=common,techniqueCategoryId=Rudder Internal,techniqueCategoryId=Active Techniques,ou=Rudder,cn=rudder-configuratio
+   * - rule: ruleId=${RELAY_UUID}-DP,ou=Rules,ou=Rudder,cn=rudder-configuration
+   * - rule: ruleId=hasPolicyServer-${RELAY_UUID},ou=Rules,ou=Rudder,cn=rudder-configuration
+   */
+  def deleteRelaySystemObjects(policyServerId: NodeId): Box[Unit] = {
+    if(policyServerId == Constants.ROOT_POLICY_SERVER_ID) {
+      Failure("Root server coniguration elements can't be deleted")
+    } else { // we don't have specific validation to do: if the node is not a policy server, nothing will be done
+      def DN(child: String, parentDN: DN) = new DN(child+","+parentDN.toString)
+      val id = policyServerId.value
+
+// nodeGroupId=hasPolicyServer-b887aee5-f191-45cd-bb2d-9e3f5b30d06e,groupCategoryId=SystemGroups,groupCategoryId=GroupRoot,ou=Rudder,cn=rudder-configuration
+
+      for {
+        con <- ldap
+        _   <- con.delete(DN(s"nodeGroupId=hasPolicyServer-${id}", dit.GROUP.SYSTEM.dn))
+        _   <- con.delete(DN(s"ruleTarget=policyServer:${id}", dit.GROUP.SYSTEM.dn))
+        _   <- con.delete(DN(s"directiveId=${id}-distributePolicy,activeTechniqueId=distributePolicy,techniqueCategoryId=Rudder Internal", dit.ACTIVE_TECHNIQUES_LIB.dn))
+        _   <- con.delete(DN(s"directiveId=common-${id},activeTechniqueId=common,techniqueCategoryId=Rudder Internal", dit.ACTIVE_TECHNIQUES_LIB.dn))
+        _   <- con.delete(DN(s"ruleId=${id}-DP", dit.RULES.dn))
+        _   <- con.delete(DN(s"ruleId=hasPolicyServer-${id}", dit.RULES.dn))
+      } yield ()
+    }
```

Comment:
there aren't any log there - if something goes bad, we'll never know.
And a debug to know that is went fine would be great